### ISSUE TITLE
Incompatible asdf version on fresh install

### ISSUE THREAD

**Issue by lheinke** [radicamc/supreme-spoon#34]
Running the `fit_lightcurves.py` file with a fresh install of the new pipeline version (1.2.0) results in the following error:
```
Traceback (most recent call last):
  File "/home/lheinke/MIRI_Exo_GTO/JWST/1281 [HAT-P-12b]/Observation 3 [NIRISS-SOSS]/Reduction/supreme-SPOON/fit_lightcurves.py", line 25, in <module>
    from supreme_spoon import stage4
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_new/lib/python3.10/site-packages/supreme_spoon/stage4.py", line 20, in <module>
    from jwst import datamodels
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_new/lib/python3.10/site-packages/jwst/datamodels/__init__.py", line 1, in <module>
    from .model_base import JwstDataModel, DataModel
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_new/lib/python3.10/site-packages/jwst/datamodels/model_base.py", line 2, in <module>
    from stdatamodels import DataModel as _DataModel
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_new/lib/python3.10/site-packages/stdatamodels/__init__.py", line 1, in <module>
    from .model_base import DataModel
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_new/lib/python3.10/site-packages/stdatamodels/model_base.py", line 21, in <module>
    from asdf.fits_embed import AsdfInFits
ModuleNotFoundError: No module named 'asdf.fits_embed'
```
I was able to fix it by manually installing asdf version 2.15.2. Perhaps this requirement should be added to the `setup.py` file. Given that this incompatibility originates from the dependency of the jwst package on asdf, this requirement will probably not be necessary anymore once supreme-SPOON migrates to a newer version of jwst.

**Comment by lheinke** [comment]
#29 would likely fix this issue implicitly.

**Comment by lheinke** [comment]
A similar problem occurs for AstroPy:
```
Traceback (most recent call last):
  File "/home/lheinke/supreme-SPOON/Standard/run_DMS.py", line 18, in <module>
    from supreme_spoon.stage1 import run_stage1
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/supreme_spoon-1.2.0-py3.10.egg/supreme_spoon/stage1.py", line 22, in <module>
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/jwst-1.8.5-py3.10-linux-x86_64.egg/jwst/pipeline/__init__.py", line 8, in <module>
    from .calwebb_spec2 import Spec2Pipeline
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/jwst-1.8.5-py3.10-linux-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 21, in <module>
    from ..residual_fringe import residual_fringe_step
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/jwst-1.8.5-py3.10-linux-x86_64.egg/jwst/residual_fringe/__init__.py", line 1, in <module>
    from .residual_fringe_step import ResidualFringeStep
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/jwst-1.8.5-py3.10-linux-x86_64.egg/jwst/residual_fringe/residual_fringe_step.py", line 4, in <module>
    from . import residual_fringe
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/jwst-1.8.5-py3.10-linux-x86_64.egg/jwst/residual_fringe/residual_fringe.py", line 12, in <module>
    from . import utils
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/jwst-1.8.5-py3.10-linux-x86_64.egg/jwst/residual_fringe/utils.py", line 16, in <module>
    from .fitter import ChiSqOutlierRejectionFitter
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/jwst-1.8.5-py3.10-linux-x86_64.egg/jwst/residual_fringe/fitter.py", line 3, in <module>
    from astropy.modeling.fitting import _model_to_fit_params
ImportError: cannot import name '_model_to_fit_params' from 'astropy.modeling.fitting' (/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/astropy-6.0.0-py3.10-linux-x86_64.egg/astropy/modeling/fitting.py)
```
This can be fixed by installing the previous version of AstroPy via `pip install astropy==5.3.4`.

Another similar issue can be encountered when trying to run the Stage 1 JumpStep:
```
Traceback (most recent call last):
  File "/home/lheinke/supreme-SPOON/Standard/run_DMS.py", line 89, in <module>
    stage1_results = run_stage1(input_files, background_model=background_model,
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/supreme_spoon-1.2.0-py3.10.egg/supreme_spoon/stage1.py", line 1376, in run_stage1
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/supreme_spoon-1.2.0-py3.10.egg/supreme_spoon/stage1.py", line 451, in run
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/stpipe-0.5.1-py3.10.egg/stpipe/step.py", line 651, in call
    instance = cls.from_config_section(config, name=name, config_file=config_file)
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/stpipe-0.5.1-py3.10.egg/stpipe/step.py", line 270, in from_config_section
    config_parser.validate(config, spec, root_dir=dirname(config_file or ""))
  File "/home/lheinke/.local/miniconda3/envs/supreme_spoon_dev/lib/python3.10/site-packages/stpipe-0.5.1-py3.10.egg/stpipe/config_parser.py", line 377, in validate
    raise ValidationError("\n".join(messages))
stpipe.config_parser.ValidationError: Extra value 'edge_size' in root
```
This required me to update to a newer (unsupported) version of the jwst pipeline.

**Comment by radicamc** [comment]
Yes, I am aware of these issues and will try to get a patch ready for this and #29 before the end of the year.
In case it helps, the local versions of these packages that I am using is `asdf==2.14.4` and `astropy==5.2.2`.
The JumpStep error comes from an update STScI made to the corresponding pars file, which makes it incompatible with older pipeline versions. Workaround are either updating the pipeline (which I have not yet tested) or a hack of renaming an older pars file. I've been using jwst_niriss_pars-jumpstep_0020.asdf, which can be downloaded from the crds here: https://jwst-crds.stsci.edu. I just renamed that file to jwst_niriss_pars-jumpstep_0081.asdf (the current most recent version), which fixes the error.

**Comment by radicamc** [comment]
Should be fixed in #37.
